core/primitives/src/stateless_validation.rs

```diff
@@ -219,6 +219,24 @@ pub struct StoredChunkStateTransitionData {
     pub receipts_hash: CryptoHash,
 }
 
+#[derive(Debug)]
+pub struct EndorsementStats {
+    pub total_stake: Balance,
+    pub endorsed_stake: Balance,
+    pub total_validators_count: usize,
+    pub endorsed_validators_count: usize,
+}
+
+impl EndorsementStats {
+    pub fn has_enough_stake(&self) -> bool {
+        self.endorsed_stake > self.required_stake()
+    }
+
+    pub fn required_stake(&self) -> Balance {
```

Comment:
nit: the naming may be a bit confusing because "required stake" hints that the minimum must be this stake, but in reality it's that the stake must be larger than this. Maybe add 1 here and use >= above?

Reply:
yeah, good point, updated!